common/src/main/java/com/microsoft/identity/common/adal/internal/net/DefaultConnectionService.java

```diff
@@ -107,10 +109,12 @@ public void onLost(@NonNull Network network) {
      */
     @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
     public void unregisterNetworkCallback() {
-        final ConnectivityManager connectivityManager = (ConnectivityManager) mConnectionContext
-                .getSystemService(Context.CONNECTIVITY_SERVICE);
+        if (networkCallback != null) {
+            final ConnectivityManager connectivityManager = (ConnectivityManager) mConnectionContext
+                    .getSystemService(Context.CONNECTIVITY_SERVICE);
 
-        connectivityManager.unregisterNetworkCallback(networkCallback);
+            connectivityManager.unregisterNetworkCallback(networkCallback);
```

Comment:
do we need to `null` out `networkCallback` too?

Reply:
Yes, I think we do. No need for it to be dangling in memory.